docs/source/examples/sky-serve.rst

```diff
@@ -0,0 +1,333 @@
+
+Hello, Sky Serve!
+-----------------
+
+Here we will go through an example to deploy a simple HTTP server with Sky Serve. To spin up a service, you can simply reuse your task YAML with the two following requirements:
+
+#. An HTTP endpoint and the port on which it listens;
+#. An extra :code:`service` section in your task YAML to describe the service configuration.
+
+It is recommended to test it with :code:`sky launch` first. For example, we have the following task YAML works with :code:`sky launch`:
+
+.. code-block:: yaml
+
+    resources:
+      ports: 8080
+      cpus: 2
+
+    workdir: .
+
+    run: python -m http.server 8080
+
+And under the same directory, we have an :code:`index.html`:
+
+.. code-block:: html
+
+    <html>
+    <head>
+        <title>My First Sky Serve Service</title>
+    </head>
+    <body>
+        <p>Hello, Sky Serve!</p>
+    </body>
+    </html>
+
+
+.. note::
+
+  :ref:`workdir <sync-code-artifacts>` and :ref:`file mounts with local files <sync-code-artifacts>` will be automatically uploaded to
+  :ref:`SkyPilot Storage <sky-storage>`. Cloud bucket will be created, and cleaned up after the service is terminated.
+
+Notice that task YAML already have a running HTTP endpoint at 8080, and exposed through the :code:`ports` section under :code:`resources`. Suppose we want to scale it into multiple replicas across multiple regions/clouds with Sky Serve. We can simply add a :code:`service` section to the YAML:
+
+.. code-block:: yaml
+
+    # hello-sky-serve.yaml
+    service:
+      readiness_probe: /
+      replicas: 2
+
+    resources:
+      ports: 8080
+      cpus: 2
+
+    workdir: .
+
+    run: python -m http.server 8080
+
+You can find more configurations in :ref:`here <service-yaml-spec>`. This example will spin up two replicas of the service, each listening on port 8080. The service is considered ready when it responds to :code:`GET /health` with a 200 status code. You can customize the readiness probe by specifying a different path in the :code:`readiness_probe` field. By calling:
+
+.. code-block:: console
+
+    $ sky serve up hello-sky-serve.yaml
+
+Sky Serve will start a centralized controller/load balancer and deploy the service to the cloud with the best price/performance ratio. It will also monitor the service status and re-launch a new replica if one of them fails.
+
+Under the hood, :code:`sky serve up`:
+
+#. Launches a controller which handles autoscaling, monitoring and load balancing;
+#. Returns a Service Endpoint which will be used to accept traffic;
+#. Meanwhile, the controller provisions replica VMs which later run the services;
+#. Once any replica is ready, the requests sent to the Service Endpoint will be **HTTP-redirect** to one of the endpoint replicas.
+
+After the controller is provisioned, you'll see the following in :code:`sky serve status` output:
+
+.. image:: ../images/sky-serve-status-output-provisioning.png
```

Comment:
One problem to fix throughout this file: the para immediately after an image becomes too hard to read due to no vertical space.